File: library-domain/src/main/java/com/dodecaedro/library/domain/repository/UserRepository.java
package com.dodecaedro.library.domain.repository;

import com.dodecaedro.library.domain.model.User;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface UserRepository {
	Optional<User> findById(UUID userId);

	User save(User user);

	List<User> findAll();

	void delete(User user);
}

File: library-domain/src/main/kotlin/com/dodecaedro/library/domain/repository/UserRepository.kt
package com.dodecaedro.library.domain.repository

import com.dodecaedro.library.domain.model.User
import java.util.*

interface UserRepository {
	fun findById(userId: UUID): Optional<User>

	fun save(user: User): User

	fun findAll(): List<User>

	fun delete(user: User)
}
